sprocs/assessment_instances_select_log.sql

```diff
@@ -340,7 +340,14 @@ BEGIN
                 WHERE
                     pvl.assessment_instance_id = ai_id
                     AND pvl.page_type = 'studentInstanceQuestion'
-                    AND pvl.authn_user_id = ai.user_id
+                    AND (pvl.authn_user_id = ai.user_id
+                         OR (ai.group_id IS NOT NULL
+                             AND EXISTS (SELECT 1
+                                         FROM group_logs AS gl
+                                         WHERE gl.action = 'join'
+                                               AND gl.group_id = ai.group_id
+                                               AND gl.user_id = pvl.authn_user_id
+                                               AND gl.date <= pvl.date)))
```

Comment:
Can you explain why this is necessary?

Reply:
If you mean the date, it is not strictly necessary, it would probably only make a difference for someone that has access to the assessment but is not part of the group (e.g., a staff member), and in that case only if that user then becomes part of the group, which presumably means this group is for staff testing only (which probably would be a bit fuzzy to determine if their view should be included). The main reason I kept it was in the spirit of "was a member of the group at the time of the view action", so "time" came into account. I'd be perfectly fine if you feel strongly about removing it. Is that what you were asking about?

Reply:
Yep, I was referring to the date. I figured it was something like that, but I had the same "hmmm..." thinking about this only really making sense if someone viewed it without being part of the group and then later joined the group. I don't feel particularly strongly about this (my PR approval stands), but I'm inclined to remove it because a) it seems to be here to support an unlikely edge case and b) I don't think it handles the case where someone (realistically, a staff member) joins a group, then leaves it, then views again. Rather than going down the edge case rabbit hole, I'm pretty comfortable saying "if user X was part of a group at any point, include their views".

All that said, good call on using `group_logs` instead of just `group_users` to ensure the case of someone leaving a group is handled correctly!

Reply:
Done. I also added a comment to clarify the filter.